python/taichi/_main.py

```diff
@@ -166,14 +166,20 @@ def gallery(self, argumets: list = sys.argv[2:]):
             "mpm128", "pbf2d", "mass_spring_game"
         ]
 
+        def valid_mouse_position(mou_x, mou_y):
+            xmin = left_margin / width
+            xmax = 1 - xmin
+            ymin = bottom_margin / height
+            ymax = 1 - horizontal_margin / height
+            return (xmin <= mou_x <= xmax) and (ymin <= mou_y <= ymax)
+
         def get_tile_from_mouse(mou_x, mou_y):
             """Find the image tile that the mouse is hovering over."""
             x = int(mou_x * width)
             y = int(mou_y * height)
             rind = (y - bottom_margin) // (vertical_margin + tile_size)
```

Comment:
```suggestion
            rind = y // (vertical_margin + tile_size)
```

This gives better experience since hovering on the text now leads to the right bounding box.